protocols/imap/src/main/java/org/apache/james/imap/decode/ImapRequestLineReader.java

```diff
@@ -814,6 +815,38 @@ private long parseUnsignedInteger(String value) throws DecodingException {
         boolean isValid(char chr);
     }
 
+    /**
+     * Verifies subsequent characters match a specified string
+     */
+    public static class StringValidator implements CharacterValidator {
+        public static StringValidator caseIncentive(String expectedString) {
+            return new StringValidator(expectedString,
+                (c1, c2) -> isaBoolean(c1, c2));
+        }
+
+        public static boolean isaBoolean(Character c1, Character c2) {
```

Comment:
I don't get what this method do by its name.
What about `asciiEqualsIgnoringCase` ?